sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -415,50 +447,74 @@ def _get_small_trainset(self, X_binned_train, y_train, seed):
                                stratify=stratify)
             X_binned_small_train = X_binned_train[indices]
             y_small_train = y_train[indices]
+            if sample_weight_train is not None:
+                sample_weight_small_train = sample_weight_train[indices]
+            else:
+                sample_weight_small_train = None
             X_binned_small_train = np.ascontiguousarray(X_binned_small_train)
-            return X_binned_small_train, y_small_train
+            return (X_binned_small_train, y_small_train,
+                    sample_weight_small_train)
         else:
-            return X_binned_train, y_train
+            return X_binned_train, y_train, sample_weight_train
 
     def _check_early_stopping_scorer(self, X_binned_small_train, y_small_train,
-                                     X_binned_val, y_val):
+                                     sample_weight_small_train,
+                                     X_binned_val, y_val, sample_weight_val):
         """Check if fitting should be early-stopped based on scorer.
 
         Scores are computed on validation data or on training data.
         """
         if is_classifier(self):
             y_small_train = self.classes_[y_small_train.astype(int)]
-        self.train_score_.append(
-            self.scorer_(self, X_binned_small_train, y_small_train)
-        )
+
+        # TODO: handle when _scorer doesn't accept sample_weight, but
+        # sample_weight is provided
+        if sample_weight_small_train is None:
+            self.train_score_.append(
+                self.scorer_(self, X_binned_small_train, y_small_train)
+            )
+        else:
+            self.train_score_.append(
+                self.scorer_(self, X_binned_small_train, y_small_train,
+                             sample_weight_small_train)
+            )
 
         if self._use_validation_data:
             if is_classifier(self):
                 y_val = self.classes_[y_val.astype(int)]
-            self.validation_score_.append(
-                self.scorer_(self, X_binned_val, y_val)
-            )
+            if sample_weight_val is None:
```

Comment:
same here about fusionning the calls

Reply:
How would you do that? I guess I'm not understanding what you mean.